### DISCUSSION TITLE
Foundry Events

### DISCUSSION THREAD

**Discussion by Niferu** [Cyfrin/foundry-full-course-cu#312] [Q&A] [answered]
Hello guys,

Here i come with another question regarding foundry as it is not explained in documentation:

Let's say we have following event:
```solidity
event TestEvent(string someString, uint256 indexed someNumber, address indexed someAddress, string someOtherString);
```

To read values emitted by that we can do following:
```solidity
contract someContract is StdCheats, Test {
    event TestEvent(string someString, uint256 indexed someNumber, address indexed someAddress, string someOtherString);

    function testEventEmits() public {
        string memory someString;
        bytes32 someNumber;
        address someAddress;
        string memory someOtherString;

        vm.expectEmit(false, true, true, false, address(contract));
        emit TestEvent(someString, uint256(someNumber), someAddress, someOtherString);

        vm.recordLogs();
        contract.someFunctionWithThatEvent("");
        Vm.Log[] memory entries = vm.getRecordedLogs();

        /// @dev Checking Test Event (emitted event is having index 2 thats why we use entries[2])
        someString = abi.decode(entries[2].data, (string));
        someNumber = entries[2].topics[1];
        someAddress = address(uint160(uint256(entries[2].topics[2])));
        someOtherString = ???; // -> if we do it as for "someString" output will be the same obviously

        console.log("String: ", someString);
        console.log("Number: ", uint256(someNumber));
        console.log("Address: ", someAddress);
        console.log("Other String: ", someOtherString);
    }
}
```

The question is how can we get that second string value as `.data` does not use indexes. Is it even possible? @PatrickAlphaC was asking similar question about events [here](https://ethereum.stackexchange.com/questions/144164/how-do-i-access-an-event-in-a-foundry-script/152314#152314) but I couldnt find answer if we can read multiple data fields of same type. Does anyone know the answer?

**Comment by shikhar229169** [comment, marked as answer]
Yes, for sure it is possible to get the value.
First you need to check what are the non-indexed members in that event.
So here we have 2, and both are string.
So, you can write like this:
```solidity
(string memory s1, string memory s2) = abi.decode(entries[2].data, (string, string));
```
In s1, you will get the ```someString``` and in s2 you will get the ```someOtherString```

**Reply by Niferu** [reply]
That is exactly what I was looking for, thanks a lot Sir!
